### ISSUE TITLE
Turning a vehicle when traversing a ramp onto a raised bridge removes part of the vehicle.

### ISSUE THREAD

**Issue by Relgar** [cataclysmbnteam/Cataclysm-BN#1212]
# Describe the bug
When your vehicle is halfway getting onto a ramp and makes a turn, the turning part of the vehicle is completely removed from the game window. When you examine the vehicle on the vehicle menu, however, it is still treated as though the tile is there. Trying to remove the 'missing' parts then results in an error

# Steps To Reproduce
1. Go up a ramp with a vehicle
2. While the vehicle hasn't cleared the ramp, make a turn on the ramp
3. Tiles are removed from the vehicle

# Expected behavior
The whole vehicle makes it onto the ramp safely

# Screenshots
![image](https://user-images.githubusercontent.com/84274761/147801583-057a91e8-3fa8-4c59-8ab6-02ae1a937101.png)

![image](https://user-images.githubusercontent.com/84274761/147801586-1851375c-c8a1-4d3c-bb40-125026199916.png)

![image](https://user-images.githubusercontent.com/84274761/147801589-fae21bb0-8bfc-4743-af5e-e1a03ed55b7a.png)

**Comment by Zireael07** [comment]
CDDA had the same bug IIRC, and I believe they managed to fix it?

**Comment by Relgar** [comment]
This sounds like the issue : https://github.com/CleverRaven/Cataclysm-DDA/pull/45726

**Comment by olanti-p** [comment]
> This sounds like the issue : CleverRaven/Cataclysm-DDA#45726

More like https://github.com/CleverRaven/Cataclysm-DDA/issues/42733, which is still not fixed.

I saw it back when I was porting bridge stuff, and ported [the band-aid](https://github.com/CleverRaven/Cataclysm-DDA/pull/47236) that should prevent you from being able to `2. While the vehicle hasn't cleared the ramp, make a turn on the ramp`, but apparently the band-aid itself (or my port) has a hole too.

That said, I just tested and [the bug](https://github.com/CleverRaven/Cataclysm-DDA/issues/44534) that CleverRaven/Cataclysm-DDA#45726 had fixed affects BN as well, so it needs to be ported either way.
